File: src/main/java/org/scalatest/spi/location/MethodFinder.java
package org.scalatest.spi.location;

public class MethodFinder implements Finder {

    public Test find(AstNode node) {
        if(node instanceof MethodDefinition) {
            MethodDefinition methodDef = (MethodDefinition) node;
            if(methodDef.getParamTypes().length == 0)
                return new Test(methodDef.getClassName(), methodDef.getClassName() + "." + methodDef.getName(), new String[] { methodDef.getName() });
            else
                return null;
        }
        else
            return null;
    }

}

File: src/main/scala/org/scalatest/spi/location/MethodFinder.scala
package org.scalatest.spi.location
import scala.annotation.tailrec

class MethodFinder extends Finder {

  @tailrec
  final def find(node: AstNode): Option[Selection] = {
    node match {
      case MethodDefinition(className, parent, children, name, paramTypes) 
        if parent != null && parent.isInstanceOf[ConstructorBlock] && paramTypes.length == 0 =>
          Some(new Selection(className, className + "." + name, Array(name)))
      case _ => 
        if (node.parent != null)
          find(node.parent)
        else
          None
    }
  }
  
}